eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,55 @@
+function Get-javascript-OnboardedDocsMsPackages($DocRepoLocation) {
+  $packageOnboardingFiles = @(
+    "$DocRepoLocation/ci-configs/packages-latest.json",
+    "$DocRepoLocation/ci-configs/packages-preview.json",
+    "$DocRepoLocation/ci-configs/packages-legacy.json")
+
+  $onboardedPackages = @{}
+  foreach ($file in $packageOnboardingFiles) {
+    $onboardingSpec = ConvertFrom-Json (Get-Content $file -Raw)
+    foreach ($spec in $onboardingSpec.npm_package_sources) {
+      $packageName = $spec.name
+      if ($packageName.LastIndexOf('@') -gt 0) {
+        # Package has an '@' symbol deliminting the end of the package name
+        $packageName = $packageName.Substring(0, $packageName.LastIndexOf('@'))
+      }
+      $onboardedPackages[$packageName] = $null
+    }
+  }
+
+  return $onboardedPackages
+}
+
+function Get-javascript-DocsMsTocData($packageMetadata, $docRepoLocation) {
+  # Fallback to get package-level readme name if metadata file info does not exist
+  $packageLevelReadmeName = $packageMetadata.Package.Replace('@azure/', '').Replace('@azure-tools/', '').Replace('azure-', '');
+
+  # Fallback to get package-level readme name if metadata file info does not exist
+  if ($packageMetadata.Package.StartsWith('@azure-rest/')) {
```

Comment:
It is unfortunate that we another place to special case azure-rest. @praveenkuttappan something to look at with your other clean-up of that processing.